### PR TITLE
Staff w21 add storybook

### PR DESCRIPTION
**Pull request by pconrad**
In this PR, we add support for Storybook, which enables us to interactively view and test React Components in isolation from other parts of the application.

To add Storybook support to an existing Spring/React project, cd into `javascript` and then run `npx sb init`.   Most of the files in this PR come from that process.

We also customized `javascript/.storybook/preview.js` to pull in the CSS files so that components appear styled in a similar way to how they will look in the actual application.

Finally, we removed the default stories and replaced them with an example of stories for the `BasicCourseTable.js` component.  This involved factoring out some text fixtures that were hard coded into the tests for BasicCourseTable, and putting them in a file that can be shared between the stories and the tests.

### CODE REVIEW

**Review by btk5h — DISMISSED** [review]

File: javascript/package.json [hunk 1]
```diff
@@ -26,7 +26,9 @@
     "build": "react-scripts build",
     "test": "react-scripts test --env=jsdom-fourteen",
     "coverage": "react-scripts test --env=jsdom-fourteen --coverage --watchAll=false",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "storybook": "start-storybook -p 6006 -s public",
+    "build-storybook": "build-storybook -s public"
```
Comment (andrewhlu) [inline javascript/package.json:31]: We might want to add the output `public` directory into the `.gitignore` file so we don't accidentally check in a static storybook into our repo.

**Review by andrewhlu — CHANGES_REQUESTED** [review]
One small suggestion

File: javascript/package.json (hunk 1 above)
Comment (pconrad) [reply javascript/package.json:31]: Done

**Review by btk5h — APPROVED** [review]

**Review by andrewhlu — APPROVED** [review]
Looks good, thanks for making the change!

**Pull request merged**